packages/backend/src/open_payments/payment/incoming/routes.ts

```diff
@@ -75,7 +76,11 @@ async function getIncomingPayment(
   }
   if (!incomingPayment) return ctx.throw(404)
   const connection = deps.connectionService.get(incomingPayment)
-  ctx.body = await incomingPaymentToBody(incomingPayment, connection)
+  ctx.body = incomingPaymentToBody(
```

Comment:
Could these `*toBody` calls be replaced with calling `*.toOpenPaymentsType` directly?

Reply:
:thinking: 
...or we do the `*withPaymentPointer` subresource type idea and make `*toBody` the limited places where we're doing the `*.paymentPointer = ctx.paymentPointer` assignment.

Reply:
>  I think I like it.
>
> I bet this would even let us remove some .withGraphFetched['paymentPointer']s in subresource service queries.

It's a little more straightforward than the alternatives, and will indeed allow us to  remove the `withGraphFetched` calls. I can do that as a separate follow-up PR

Reply:
I don't feeling too strongly about `toBody` or `*withPaymentPointer`.
The latter would let `getUrl` be a `url` getter but it'd probably only be called within `toOpenPaymentType`.

Reply:
I guess `url` getter would also be used in tests...